container/app/entrypoint.py

```diff
@@ -0,0 +1,70 @@
+#!/venv/bin/python3
+#
+# SPDX-FileCopyrightText: Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import argparse
+import json
+import os
+import subprocess
+
+from ovirt_imageio.client import _utils
+from ovirt_imageio._internal import qemu_img
+from ovirt_imageio._internal import qemu_nbd
+from ovirt_imageio._internal import nbd
+
+TICKET_PATH = "/app/ticket.json"
+OVIRT_IMAGEIO = "/venv/bin/ovirt-imageio"
+
+
+def _generate_ticket(uuid, socket, size):
+    ticket_info = {}
+    ticket_info["uuid"] = uuid
+    ticket_info["url"] = f"nbd:unix:{socket}"
+    ticket_info["size"] = size
+    ticket_info["timeout"] = 3000
+    ticket_info["inactivity_timeout"] = 300
+    ticket_info["ops"] = ["read"]
+    return ticket_info
+
+
+def create_ticket(ticket_id, image, socket):
+    image_size = qemu_img.info(image)["virtual-size"]
+    ticket_info = _generate_ticket(ticket_id, socket, image_size)
+    with open(TICKET_PATH, "w+") as f:
+        f.write(json.dumps(ticket_info))
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="ovirt-imageio")
+    parser.add_argument("image", help="Path to image.")
+    parser.add_argument(
+        "--ticket-id",
+        help="Optional. Set the ID of the ticket. It is recommended "
+        "to use with random string for better security. "
+        "Defaults to the image filename.")
+
+    return parser.parse_args()
+
+
+def main():
+    args = _parse_args()
+    image_format = qemu_img.info(args.image)["format"]
```

Comment:
We need the image virtual-size when creating the ticket. You run qemu-img gain there, but this is not free and add unwanted delay. The right way is to run qemu-img once and pass the return value to the reset of the code.

We can do:

    image_info = image_format = qemu_img.info(args.image)
    ...
        create_ticket(ticket_id, image_info, socket_path)
        ...
        with qemu_nbd.run(args.image, image_info["format"], sock):